contracts/TransmuteDPOS.sol

```diff
@@ -76,11 +65,13 @@ contract TransmuteDPOS is TransmuteToken, RoundManager, DelegatorManager {
     // Sets the block number from which the Delegator will be able to withdraw() his tokens
     uint withdrawBlock = block.number.add(unbondingPeriod);
     uint amount = d.amountBonded;
-    withdrawInformations[msg.sender] = WithdrawInformation(withdrawBlock, amount);
+    unbondingInformations[msg.sender] = UnbondingInformation(withdrawBlock, amount);
     if (d.delegateAddress == msg.sender) {
-      // A Provider has to be a Delegator to himself
-      // Therefore if a Provider unbonds he should resign
-      resignAsProvider(msg.sender);
+      if (providerStatus(msg.sender) == ProviderStatus.Registered) {
+        // A Provider has to be a Delegator to himself
+        // Therefore if a Provider unbonds he should resign
+        resignAsProvider(msg.sender);
```

Comment:
I wonder about the logic of this. It could be the case that we throw here, and require the provider to call resign directly. interpreting the user intention seems worse then forcing them to follow a specific order of operations.... is there are counter point?

Reply:
If a Provider unbonds, he breaks rule number 1 of being a Provider: "A Provider must bond to himself before registering". Therefore if he unbonds, he is by definition not a Provider anymore (or else all the code that relies on rule number 1 would break).

I agree that my comment is misleading. It's not like he should resign and the contract is making the decision for him. It's that he isn't a Provider anymore.